oneview/resource_server_hardware.go

```diff
@@ -200,14 +202,32 @@ func resourceServerHardwareCreate(d *schema.ResourceData, meta interface{}) erro
 }
 
 func resourceServerHardwareRead(d *schema.ResourceData, meta interface{}) error {
+	var (
+		servHard ov.ServerHardware
+		err      error
+	)
 	config := meta.(*Config)
 
-	servHard, err := config.ovClient.GetServerHardwareByName(d.Get("hostname").(string))
+	// fetching server hardware hostname incase it's added
+	if val, ok := d.GetOk("hostname"); ok {
+		servHard, err = config.ovClient.GetServerHardwareByName(val.(string))
+	} else {
+		// for refreshing imported server hardware we would need it's name
+		if val, ok := d.GetOk("name"); ok {
+			servHard, err = config.ovClient.GetServerHardwareByName(val.(string))
+		} else {
+			// for importing server hardware
+			servHard, err = config.ovClient.GetServerHardwareByName(d.Id())
+		}
+	}
+
 	if err != nil || servHard.URI.IsNil() {
 		d.SetId("")
-		return nil
+		return fmt.Errorf("unable to retrieve server hardware %s", err)
 	}
-	d.SetId(d.Id())
+	// setting UUID as resource Id
+	d.SetId(servHard.UUID.String())
```

Comment:
looks like we are setting uuid as id rather than name...have we looked into it clearly...because if we set uuid as id then not sure how GetServerHardwareByName(d.id()) will work?